packages/iTwinUI-react/src/core/utils/hooks/useDragAndDrop.tsx

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+/**
+ * Helper hook that handles elements drag logic.
+ * @param elementRef Element ref that is draggable.
+ * @returns
+ * `onPointerDown` - handler that is called when pointer is down and handles all the dragging logic.
+ * `transform` - current transform of the element, it is used to preserve drag position when element visibility is being toggled.
+ */
+export const useDragAndDrop = (elementRef: React.RefObject<HTMLElement>) => {
+  const grabOffsetX = React.useRef(0);
+  const grabOffsetY = React.useRef(0);
+  const translateX = React.useRef(0);
+  const translateY = React.useRef(0);
+
+  const [transform, setTransform] = React.useState('');
+
+  const onPointerMove = React.useCallback(
+    (event: PointerEvent) => {
+      if (elementRef.current) {
+        translateX.current = event.clientX - grabOffsetX.current;
+        translateY.current = event.clientY - grabOffsetY.current;
+        elementRef.current.style.transform = `translate(${translateX.current}px, ${translateY.current}px)`;
+      }
+    },
+    [elementRef],
```

Comment:
I know its linter suggesting it here... but maybe it would make sense to accept `HTMLElement | null` instead of the ref..?

Reply:
What it would give? I am afraid it will make too many unnecessary rerenders because of element changes.

Reply:
just whenever see ref in deps list, looks like something is wrong..
but I guess leave it, as it works fine.